test/python/result/test_mitigators.py

```diff
@@ -119,8 +119,8 @@ def counts_data(circuit, assignment_matrices, shots=1024):
     def test_mitigation_improvement(self):
         """Test whether readout mitigation led to more accurate results"""
         shots = 1024
-        assignment_matrices = self.assignment_matrices()
-        num_qubits = len(assignment_matrices)
+        with self.assertWarns(DeprecationWarning):
+            assignment_matrices = self.assignment_matrices()
```

Comment:
What triggers a warning in this test?

Reply:
good catch! I added a comment in 889173bc154b5fe79594694a2fadd8da59986753 and submitted https://github.com/Qiskit/qiskit/issues/12832